user-manager/src/main/java/com/qpay/usermanager/services/impl/MerchantServiceImpl.java

```diff
@@ -0,0 +1,75 @@
+package com.qpay.usermanager.services.impl;
+
+import com.qpay.usermanager.mappers.MerchantMapper;
+import com.qpay.usermanager.model.dto.merchant.MerchantCreation;
+import com.qpay.usermanager.model.entity.merchant.MerchantEntity;
+import com.qpay.usermanager.repositories.MerchantRepository;
+import com.qpay.usermanager.services.MerchantService;
+import com.qpay.usermanager.services.exception.NoMerchantFoundException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class MerchantServiceImpl implements MerchantService {
+
+    private static final String GET_MERCHANT_BY_ID_ERROR = "Board with given id doesn't exist";
+
+    private final MerchantRepository merchantRepository;
+
+    private final MerchantMapper merchantMapper;
+
+    @Override
+    public List<MerchantEntity> getMerchants() {
+        return merchantRepository.findMerchantEntitiesByOrderByName();
+    }
+
+    @Override
+    public MerchantEntity getMerchantById(final long id) {
+        final Optional<MerchantEntity> merchantEntityOptional = merchantRepository.findById(id);
+        if (merchantEntityOptional.isEmpty()) {
+            log.error(GET_MERCHANT_BY_ID_ERROR);
+            throw new NoMerchantFoundException(GET_MERCHANT_BY_ID_ERROR);
+        }
+
+        return merchantEntityOptional.get();
+    }
+
+    @Override
+    public MerchantEntity addMerchant(final MerchantCreation merchantCreation) {
+        final MerchantEntity merchantEntity = merchantMapper.map(merchantCreation);
+        merchantRepository.save(merchantEntity);
+        return merchantEntity;
+    }
+
+    @Override
+    public void deleteMerchant(final long id) {
+        if (!merchantRepository.existsById(id)) {
+            log.error(GET_MERCHANT_BY_ID_ERROR);
+            throw new NoMerchantFoundException(GET_MERCHANT_BY_ID_ERROR);
+        }
+        merchantRepository.deleteById(id);
```

Comment:
what if you try to call this method `merchantRepository.deleteById(id);` when id does not exist. What will you get?